translate/storage/jsonl10n.py

```diff
@@ -416,6 +416,94 @@ def _extract_units(
             yield from parent
 
 
+class GoTextJsonUnit(BaseJsonUnit):
+    ID_FORMAT = "{}"
+
+    def getvalue(self):
+        target = self.target
+        if isinstance(target, multistring):
+            strings = list(target.strings)
+            if len(self._store.plural_tags) > len(target.strings):
+                strings += [""] * (len(self._store.plural_tags) - len(target.strings))
+            target = {
+                "select": {
+                    "feature": "plural",
+                    "arg": "N",
+                    "cases": {
+                        plural: {"msg": strings[offset]}
+                        for offset, plural in enumerate(self._store.plural_tags)
+                    },
+                }
+            }
+        value = {"id": self.getid()}
+        if self.notes:
+            value["translatorComment"] = self.notes
+        value["translation"] = target
+        return value
+
+
+class GoTextJsonFile(JsonFile):
+    """gotext JSON file
+
+    See following URLs for doc:
+
+    https://pkg.go.dev/golang.org/x/text/cmd/gotext
+    https://github.com/golang/text/tree/master/cmd/gotext/examples/extract/locales/en-US
+    """
+
+    UnitClass = GoTextJsonUnit
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        locale = self.gettargetlanguage()
+        if locale:
+            self.locale = locale.replace("_", "-").split("-")[0]
+        else:
+            self.locale = "en"
+
+    @property
+    def plural_tags(self):
+        return plural_tags.get(self.locale, plural_tags["en"])
+
+    def _extract_units(
+        self,
+        data,
+        stop=None,
+        prev=None,
+        name_node=None,
+        name_last_node=None,
+        last_node=None,
+    ):
+        for value in data:
```

Comment:
Shouldn't this process only messages?

```suggestion
        for value in data["messages"]:
```

Just guessing, as I saw the test failure.

Reply:
Fixed, thanks!